src/Manager/Environment/EnvironmentMenu.jsx

```diff
@@ -0,0 +1,92 @@
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import './style.scss';
+
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { exec } from 'child_process';
+import path from 'path';
+import { shell } from 'electron';
+import Dropdown from 'react-bootstrap/Dropdown';
+import DropdownButton from 'react-bootstrap/DropdownButton';
+import { cloneNcs } from './environmentEffects';
+import { isInstalled, toolchainDir, version } from './environmentReducer';
+import { showConfirmInstallDirDialog } from '../../InstallDir/installDirReducer';
+import { showConfirmRemoveDialog } from '../managerReducer';
+
+import environmentPropType from './environmentPropType';
+
+const openBash = folder => {
+    exec(`"${path.resolve(folder, 'git-bash.exe')}"`);
+};
+
+const openCmd = folder => {
+    exec(`start cmd /k "${path.resolve(folder, 'git-cmd.cmd')}"`);
+};
+
+const openFolder = folder => {
+    shell.openItem(folder);
+};
+
+const EnvironmentMenu = ({ environment }) => {
+    const dispatch = useDispatch();
+    return (
+        <DropdownButton
+            className="ml-2"
+            variant="outline-primary"
+            title=""
+            alignRight
+            disabled={!isInstalled(environment)}
+        >
+            {/* eslint-disable max-len */}
+            <Dropdown.Item onClick={() => openBash(toolchainDir(environment))}>Open bash</Dropdown.Item>
```

Comment:
You could lift up `toolchainDir(environment)` and `version(environment)` as consts before return further simplifying and probably getting rid of eslint override.

Reply:
Extracted the consts in d987be7 but it was not sufficient for the eslint rule. 

To make the lines short enough I could have extracted the whole handler functions into consts (then I also could have used `useCallback` to memoize them) but the current code with the long lines seemed more readable to me (or [“scannable”, as I recently heard from GeePaw](https://www.geepawhill.org/2020/03/03/readability-and-scannability/)).